server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -571,17 +572,6 @@ public void testOperationPermitsOnPrimaryShards() throws InterruptedException, E
         }
         final long primaryTerm = indexShard.getPendingPrimaryTerm();
         assertEquals(0, indexShard.getActiveOperationsCount());
-        if (indexShard.routingEntry().isRelocationTarget() == false) {
-            try {
-                final PlainActionFuture<Releasable> permitAcquiredFuture = new PlainActionFuture<>();
-                indexShard.acquireReplicaOperationPermit(primaryTerm, indexShard.getGlobalCheckpoint(), permitAcquiredFuture,
```

Comment:
should we replace this with an expect assertion block?

Reply:
unfortunately does not work. The assertion is executed on another thread and the `AssertionError` can therefore not be caught here, instead it ends up in the uncaught exception handler, failing the test.

Reply:
I can suggest making it an evil test (no security manager) and attaching your own uncaught exception handler?

Reply:
I took another look at this test to see where a delay would come from (otherwise the exception should bubble up) and I think it's from the [promotion clause](https://github.com/elastic/elasticsearch/pull/32667/files/cf3322bc480b363caa7176577df3c9f23065e7a0#diff-eab7c4e55b116ae19cce00a04004e797R558) above. I wonder if we should wait for it to complete it's background work to avoid a noisy test (and also enable this part of the test)

Reply:
I've changed the test so that it waits for the promotion to complete and reenabled the check in 7ad3d25.